File: reto_clean_code/python/domain/pedido.py
"""
Representa un pedido dentro del sistema.  Contiene un identificador,
el total y un estado que puede cambiarse a confirmado o cancelado.
"""


class Pedido:
    def __init__(self, id: str, total: float) -> None:
        self.id = id
        self.total = total
        self.estado = "PENDIENTE"

    def confirmar(self) -> None:
        self.estado = "CONFIRMADO"

    def cancelar(self) -> None:
        self.estado = "CANCELADO"

File: reto_clean_code/angular/domain/pedido.ts
// Representa un pedido que puede confirmarse o cancelarse.  No incluye la
// relación con el carrito ni el pago para simplificar.

export class Pedido {
  public estado = 'PENDIENTE';

  constructor(public id: string, public total: number) {}

  confirmar(): void {
    this.estado = 'CONFIRMADO';
  }

  cancelar(): void {
    this.estado = 'CANCELADO';
  }
}